src/hooks/util.ts

```diff
@@ -8,14 +8,16 @@ export const useTappay = () => {
   const TPDirect = (window as any)['TPDirect']
   const { settings } = useApp()
 
-  settings['tappay.app_id'] &&
-    settings['tappay.app_key'] &&
-    TPDirect &&
-    TPDirect.setupSDK(
-      settings['tappay.app_id'],
-      settings['tappay.app_key'],
-      settings['tappay.dry_run'] === 'true' ? 'sandbox' : 'production',
-    )
+  useMemo(() => {
+    settings['tappay.app_id'] &&
+      settings['tappay.app_key'] &&
+      TPDirect &&
+      TPDirect.setupSDK(
+        settings['tappay.app_id'],
+        settings['tappay.app_key'],
+        settings['tappay.dry_run'] === 'true' ? 'sandbox' : 'production',
+      )
+  }, [TPDirect, settings])
```

Comment:
useMemo doesn't seem to be made for you to use like this...
ref: https://zh-hant.reactjs.org/docs/hooks-reference.html#usememo